docs/pages/docs/guides/images-and-files.mdx

```diff
@@ -0,0 +1,272 @@
+
+Getting started with Images and Files in Keystone requires you first to create a `storage` object in your Keystone configuration file. Any number of stores can be set up within the `storage` object, and you can mix and match between `local` and `s3` depending on your use case.
+
+The `storage` object defines how and where the assets are stored and accessed by both Keystone and the client frontend. This object defines:
+
+- The `kind` of storage being used - either `s3` or `local`
+- The `type` of `field` the storage is being used for - either `file` or `image`
+- A function to generate the URL (`generateUrl`) that Keystone returns in the GraphQL API pointing to the location or the storage where the assets can be accessed
+- The actual location where Keystone stores the assets - either a local `path` or `s3` bucket details
+- The location Keystone will serve the assets from - either a `serverRoute` for `local` or a `proxied` connection for `s3`. Both of these options add a route to the Keystone backend which the files can be accessed from
+
+### Defining `storage` in Keystone config
+
+First, we are going to use [dotenv](https://www.npmjs.com/package/dotenv) to retrieve the S3 bucket and URL details from a `.env` file or set environment variables.
+
+In your Keystone Config file...
+
+```typescript{24-54}
+import { config } from '@keystone-6/core';
+import dotenv from 'dotenv';
+import { lists } from './schema';
+
+dotenv.config();
+
+ const {
+  // The S3 Bucket Name used to store assets
+  S3_BUCKET_NAME: bucketName = 'keystone-test',
+  // The region of the S3 bucket
+  S3_REGION: region = 'ap-southeast-2',
+  // The Access Key ID and Secret that has read/write access to the S3 bucket
+  S3_ACCESS_KEY_ID: accessKeyId = 'keystone',
+  S3_SECRET_ACCESS_KEY: secretAccessKey = 'keystone',
+  // The base URL to serve assets from
+  ASSET_BASE_URL: baseUrl = 'http://localhost:3000,
+ } = process.env;
+/** ... */
+```
+
+!> The above pulls the environment variables and maps them to friendly names to use below
+
+### Storing your assets in `s3`
+
+We can then add an `s3` `storage` object, the object below is called `my_s3_files` and this is the name that we will use in our `field` config later. This can be called any name that makes sense to your use case.
+
+In the [config](../apis/config) object in your `keystone.ts` file...
+
+```typescript
+/** config */
+storage: {
+  my_s3_files: {
+    // Files that use this store will be stored in an s3 bucket
+    kind: 's3',
+    // This store is used for the file field type
+    type: 'file',
+    // The S3 bucket name pulled from the S3_BUCKET_NAME environment variable above
+    bucketName,
+    // The S3 bucket region pulled from the S3_REGION environment variable above
+    region,
+    // The S3 Access Key ID pulled from the S3_ACCESS_KEY_ID environment variable above
+    accessKeyId,
+    // The S3 Secret pulled from the S3_SECRET_ACCESS_KEY environment variable above
```

Comment:
```suggestion
    // The S3 bucket name pulled from the S3_BUCKET_NAME environment variable
    bucketName,
    // The S3 bucket region pulled from the S3_REGION environment variable
    region,
    // The S3 Access Key ID pulled from the S3_ACCESS_KEY_ID environment variable
    accessKeyId,
    // The S3 Secret pulled from the S3_SECRET_ACCESS_KEY environment variable
```

Removing the `above` reference from code sample in case people copy paste this in, and above/below loses context.